### PR TITLE
Two little improvements

### PR DESCRIPTION
**Pull request by vanodevium**
- removed unnecessary operation from the cycle;
- switched to decrement;

### CODE REVIEW

**Review by peter279k — APPROVED** [review]
It looks good for improving logical calculating the `$nDigits` and increase duplicate `$inputLength` variable calculating.

**Comment by vanodevium** [comment]
@peter279k 

Sorry, I don't quite understand what you mean. Could you explain in a little more detail?

File: src/Luhn/Algorithm/LuhnAlgorithm.php [hunk 1]
```diff
@@ -36,11 +36,11 @@ public function computeCheckDigit(string $input): int
...
     {
         // Need to account for the check digit.
-        $nDigits = \strlen($input) + 1;
-        $parity = $nDigits % 2;
+        $nDigits = \strlen($input);
+        $parity = ($nDigits + 1) % 2;
         $checksum = 0;
 
-        for ($i = 0; $i < $nDigits - 1; ++$i)
+        for ($i = 0; $i < $nDigits; ++$i)
```
Comment (peter279k) [inline src/Luhn/Algorithm/LuhnAlgorithm.php:43]: Here is the improvement for `$nDigit` calculating.

File: src/Luhn/Algorithm/LuhnAlgorithm.php [hunk 2]
```diff
@@ -68,10 +68,12 @@ protected function cleanAndSplitInput(string $input): array
         if ($inputLength === 0) {
             throw new InvalidArgumentException;
         }
-
+        
+        $inputLength--;
```
Comment (peter279k) [inline src/Luhn/Algorithm/LuhnAlgorithm.php:72]: Using the `$inputLength--` firstly to avoid duplicated `$inputLength-1` calculating.

**Comment by vanodevium** [comment]
@peter279k 

Do I need to change anything in this patch? :)

**Comment by peter279k** [comment]
> @peter279k
> 
> Do I need to change anything in this patch? :)

No. It looks good for me.

**Comment by vanodevium** [comment]
@peter279k 

Thanks!

**Comment by vpratfr** [comment]
Thanks. Merged in 1.0.9.

**Pull request merged**